Kitodo/src/test/java/org/kitodo/production/forms/copyprocess/CreateProcessFormIT.java

```diff
@@ -75,65 +80,132 @@ public static void cleanDatabase() throws Exception {
+        CreateProcessForm underTest = setupCreateProcessForm("Monograph");
+        underTest.getMainProcess().setTitle("title with whitespaces");
+        long before = processService.count();
+        assertThrows(ProcessGenerationException.class, underTest::createProcessHierarchy,
+                "Expected a ProcessGenerationException to be thrown for an invalid title, but it was not.");
+        long after = processService.count();
+        // Ensure no process was created
+        assertEquals(before, after, "A process with an invalid title was created!");
+    }
+
+
+    @Test
+    public void shouldNotAllowDuplicateProcessTitles() throws Exception {
+        assertDuplicateTitleNotAllowed(1, 1, false);
+    }
+
+    @Test
+    public void shouldNotAllowProcessTitlesInProjectsTheUserDoesNotBelongTo() throws Exception {
+        assertDuplicateTitleNotAllowed(2, 1, true);
+    }
 
-        assertTrue(newProcess.getTasks().isEmpty(), "Process should not have tasks");
-        assertNull(newProcess.getSortHelperStatus(), "process should not have sortHelperStatus");
+    private void assertDuplicateTitleNotAllowed(int firstProjectId, int secondProjectId, boolean switchUserContext) throws Exception {
+        // First process creation
+        CreateProcessForm underTest = setupCreateProcessForm("Monograph");
+        underTest.getMainProcess().setTitle("title");
+        underTest.getMainProcess().setProject(ServiceManager.getProjectService().getById(firstProjectId));
+
+        setScriptPermissions(true);
+        long before = processService.count();
+        underTest.createProcessHierarchy();
+        setScriptPermissions(false);
 
-        // clean up database, index and file system
-        Integer processId = newProcess.getId();
-        processService.remove(processId);
-        fileService.delete(URI.create(processId.toString()));
+        long after = processService.count();
+        createdProcess = underTest.getMainProcess();
+        assertEquals(before + 1, after, "First process creation failed. No process was created!");
+
+        // Switch user context to check with a user which does not has access to project 2
+        if (switchUserContext) {
+            User userTwo = ServiceManager.getUserService().getById(2);
+            SecurityTestUtils.addUserDataToSecurityContext(userTwo, 1);
+            // Assert that the user 2 is NOT associated with project 2
+            assertFalse(ServiceManager.getProjectService()
+                    .getById(firstProjectId)
+                    .getUsers()
+                    .contains(userTwo), "User 2 should not have access to project 2");
+        }
+        // Second process creation with duplicate title
+        CreateProcessForm underTestTwo = setupCreateProcessForm("Monograph");
+        underTestTwo.getMainProcess().setTitle("title");
+        underTestTwo.getMainProcess().setProject(ServiceManager.getProjectService().getById(secondProjectId));
```

Comment:
Before i used project 2 in both process creations during the `shouldNotAllowProcessTitlesInProjectsTheUserDoesNotBelongTo`-test. I changed that. The second process creation should happen in project 1, where user 2 has access to. The process creation should fail because of a duplicate title not because the user has not access to the project. It worked before but this makes the intent clearer.